lib/vmpooler/pool_manager.rb

```diff
@@ -746,7 +764,12 @@ def execute!
           end
         end
 
-        sleep(1)
+        sleep(loop_delay)
+
+        unless maxloop.zero?
+          break if loop_count >= maxloop
+          loop_count = loop_count + 1
+        end
       end
     end
   end
```

Comment:
The commit messages make it sound like it should all be test changes but these look like code changes. Is this expected?